crates/turbopack-tests/js/jest-entry.ts

```diff
@@ -0,0 +1,26 @@
+declare const __turbopack_external_require__: (id: string) => any;
+const jest = __turbopack_external_require__("jest-circus");
+
+const uncaughtExceptions: string[] = [];
+const unhandledRejections: string[] = [];
+
+process.on("uncaughtException", (e) => {
+  uncaughtExceptions.push(String(e));
+});
+
+process.on("unhandledRejection", (e) => {
+  unhandledRejections.push(String(e));
+});
+
+export default async function run() {
+  const jestResult = await jest.run();
+
+  // Wait a full tick for unhandledRejection handlers to run -- a microtask is not sufficient.
+  await new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
Technically, this isn't sufficient either. We'd need something like a `FinalizationRegistry` paired with an `AsyncLocalStorage` to detect that all async processes have finished.